### DISCUSSION TITLE
Standardization of commit messages

### DISCUSSION THREAD

**Discussion by delisma** [gc-da11yn/gc-da11yn.github.io#198] [Best Practices (Open Discussions)]
A well-crafted Git commit message is the best way to communicate context about a change to other developers working on the project and to your future self as well. Commit messages can adequately communicate why a change was made, and understanding that makes development and collaboration more efficient.

## How should we write commit messages
Only using git commit -m "Fix X to allow Y to use Z" with just a subject and no extra description could be great for small and clear fixes like git commit -m "Fix typo in README.md. But in cases of more extensive changes, we would need to add some extra details.

### How to write good commit messages
There are several conventions used by different teams and developers to write good commit messages. Let's outline some general rules and tips for writing commit messages–we have to decide what convention we want to follow.

1. Specifying the type of commit:
   * feat: The new feature you're adding to a particular application
   * fix: A bug fix
   * style: Feature and updates related to styling
   * refactor: Refactoring a specific section of the codebase
   * test: Everything related to testing
   * docs: Everything related to documentation
   * chore: Regular code maintenance.[ You can also use emojis to represent commit types]
2. Separating the subject from the body with a blank line
3. The commit message should not contain any whitespace errors
4. Removing unnecessary punctuation marks
5. Not ending the subject line with a period
6. Capitalizing the subject line and each paragraph
7. Using the imperative mood in the subject line
8. Using the body to explain what changes we have made and why we made them.
9. Not assuming the reviewer understands what the original problem was, ensure we add it.
10. Not thinking our code is self-explanatory
11. Follow the commit convention defined by the team

### Other practices to consider
 ![image](https://user-images.githubusercontent.com/2599251/196237444-2ea185c5-c45d-4135-a048-cda3b98ac7a9.png)

or

```
<type>: <description>

[optional body]

[optional footer]
```
or
```
JIRA-123 Concise description of the change
JIRA-123 Componentname: concise description of the change
```
or 

```
<issue-ID> <verb in present-tense> <rest of message (max 80 char)>

free text in body!
```
or
```
feature/patch/update/fix:Short description of the commit

Body: If any instructions/important info
```

## Conclusion
The benefits of writing good commit messages are not only limited to the team, but also expand to ourself and future contributors.

**Comment by shawnthompson** [comment]
I'm a fan of linking commits to issues or discussions so simple commit messages with linking to an issue #.

Maybe a simple 

Example:

```
bug: fix error in heading

fixes #7582
```
